modelling/src/main/java/org/axonframework/modelling/entity/PolymorphicEntityModel.java

```diff
@@ -104,21 +111,55 @@ private <T extends E> EntityModel<T> modelFor(T entity) {
     @Nonnull
     @Override
     public Set<QualifiedName> supportedCommands() {
-        return supportedCommands;
+        return supportedCommandNames;
+    }
+
+    @Override
+    @Nonnull
+    public Set<QualifiedName> supportedCreationalCommands() {
+        return supportedCreationalCommandNames;
```

Comment:
I suggest returning a copy of the set in order do not modify it from the outside.
The same for `supportedCommands`, `supportedCreationalCommands`, `supportedInstanceCommands`. 

Or make it package-private if possible.

Reply:
Plus 1 to this!